crypto/bn/bn_gcd.c

```diff
@@ -525,9 +292,29 @@ BIGNUM *BN_mod_inverse(BIGNUM *in,
         }
     }
 
-    rv = int_bn_mod_inverse(in, a, n, ctx, &noinv);
+    BN_CTX_start(ctx);
+
+    /* ensure n is non-negative */
+    if (BN_is_negative(n)) {
+        if ((n_pos = BN_CTX_get(ctx)) == NULL
+            || BN_copy(n_pos, n) == NULL)
+            goto err;
+        n_pos->neg = 0;
+    }
+
+    /* ensure a is non-negative */
+    if (BN_is_negative(a) && !BN_is_zero(n)) {
+        if ((a_pos = BN_CTX_get(ctx)) == NULL
+            || !BN_nnmod(a_pos, a, n, ctx))
```

Comment:
Should this conditionally use n_pos?

Reply:
I thought about it at the time, but it seems in `BN_nnmod` the sign of the modulus is a don't care. I'll double check, though.

Reply:
added as part of bad6659b1ef56c37a0887df82b0ccbdc901bc037